docs(style-guide): Added avoid contractions/possessives (#26591)
renovatebot/renovate: docs/development/style-guide.md

And so on.
```

## Avoid contractions/possessives

Avoid:

```markdown
don't
won't
doesn't
shouldn't
wouldn't
manager's
file's

> ```

Author: I've also seen `bot's` a multiple times, should I add this as well?

Reviewer: I don't think it's confusing if people say "the bot's schedule" or something like that

Reviewer 2: I agree with `@rarkins`. It's okay to use `the bot's x`. 😄

Do:

```markdown
do not
will not
does not
should not
would not
```

> Avoid possessives like `manager's` or `file's`, if needed rewrite the sentence.

Author: Maybe we could add some examples? Most times manager's and bot's can be rewritten as `Renovate's manager` and `Renovate's` respectively.

Reviewer 2: Hmm, I'm not sure about `manager's` being easy to rewrite. From a quick Renovate docs search for `manager's`, it looks like `manager's` mostly refers to any of Renovate's managers. Not to a specific "Renovate manager". I hope this makes sense.... Here are some examples of what I mean:

> Renovate searches in each repository for any files matching each manager's configured `fileMatch` pattern(s)

> #### Extending a manager's default fileMatch
> 
> If the default `fileMatch` regular expression for a manager does not match against one of your relevant files, you can _extend_ the existing regular expression(s) by configuring a manager's `fileMatch` like in this example:
> 
> ```json
> {
>   "dockerfile": {
>     "fileMatch": ["does-not-look-like-a-docker-file"]
>   }
> }
> ```

> But when such files need updating, Renovate does so by using the package managers themselves instead of trying to "reverse engineer" each package manager's file formats and behavior.

And so on.
```

## Avoid contractions

Avoid:

```markdown
don't
won't

> shouldn't

Reviewer 2: ```suggestion
doesn't
shouldn't
```

Adding `doesn't` so the order of the avoid and do list is the same.

wouldn't
manager's
file's

> Renovate's

Reviewer 2: ```suggestion
```

I'm happy to keep possessives like `Renovate's package manager` or `Renovate's style guide`.

We should probably avoid possessives like `manager's` and `file's`. With `Renovate's` its clear we refer to the bot, but with `manager's` or `file's` it's not so easy.

```

Do:

```markdown
do not
will not
does not
should not
would not
```

> For contractions like "manager's", "file's", and "Renovate's" often require rewriting the sentence.

Reviewer 2: ```suggestion
Avoid possessives like `manager's` or `file's`, if needed rewrite the sentence.
```

> Therefore, those words do not have an example listed above.

Reviewer 2: ```suggestion
```
